internal/tiltfile/tiltfile_state.go

```diff
@@ -1192,17 +1293,51 @@ const (
 
 // A selector matches an entity if all non-empty selector fields match the corresponding entity fields
 type k8sObjectSelector struct {
-	apiVersion *regexp.Regexp
-	kind       *regexp.Regexp
+	apiVersion       *regexp.Regexp
```

Comment:
this seems convoluted, would you consider something like https://github.com/windmilleng/tilt/blob/c2b2df88de3777eed5f1bb9f54b5c555707c8b42/internal/container/selector.go#L9 so you only have to store each bit of information once?

Reply:
^ or else put a TODO